pkg/podexec/pod_command_executor.go

```diff
@@ -123,6 +123,12 @@ func (e *defaultPodCommandExecutor) ExecutePodCommand(log logrus.FieldLogger, it
 			"hookTimeout":   localHook.Timeout,
 		},
 	)
+
+	if pod.Status.Phase == corev1api.PodSucceeded || pod.Status.Phase == corev1api.PodFailed {
```

Comment:
Sorry for the late response, but after going through the code again, I think we may need to change the condition check from current status into something like this:

``` go
if pod.Status.Phase !=  corev1api.PodRunning {
    ...
}
```